tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/SharePointInfo/Taxonomy.cs

```diff
@@ -391,9 +391,24 @@ private string GetManagedTermSetHierarchy(ClientContext clientContext, TermSet t
                             }
                             else if (customProperty.Key.Equals(taxonomySettings.SubAreaCustomPropertyisNoFolderStructurePresent, StringComparison.Ordinal))
                             {
-
                                 jw.WritePropertyName("isNoFolderStructurePresent");
-                                jw.WriteValue(customProperty.Value);
+                                if (generalSettings.IsBackwardCompatibale)
```

Comment:
For backward code compatibility, a new setting has been added which will decide whether to create folders inside document library when creating matter